lib/core/src/payjoin/side_swap.rs

```diff
@@ -0,0 +1,679 @@
+                )))?;
+        let Some(fiat_id) = asset_metadata.fiat_id.clone() else {
+            return Err(PayjoinError::generic(format!(
+                "No fiat ID available in asset metadata for {asset_id}"
+            )));
+        };
+        let fiat_rates = self.fiat_api.fetch_fiat_rates().await?;
+        let usd_index_price = fiat_rates
+            .iter()
+            .find(|rate| rate.coin == "USD")
+            .map(|rate| rate.value)
+            .ok_or(PayjoinError::generic("No rate available for USD"))?;
+        let asset_index_price = fiat_rates
+            .iter()
+            .find(|rate| rate.coin == fiat_id)
+            .map(|rate| rate.value)
+            .ok_or(PayjoinError::generic(format!(
+                "No rate available for {fiat_id}"
+            )))?;
+
+        let fixed_fee = (SIDESWAP_BASE_USD_FEE_SAT / usd_index_price * asset_index_price) as u64;
+        // Fees assuming we have:
+        // - 1 input for the server (lbtc)
+        // - 1 input for the user (asset)
+        // - 1 output for the user (asset change)
+        // - 1 output for the recipient (asset)
+        // - 1 output for the server (asset fee)
+        // - 1 output for the server (lbtc change)
+        let network_fee = TxFee {
+            server_inputs: 1,
+            user_inputs: 1,
+            outputs: 4,
+        }
+        .fee();
+        let fee_sat = (network_fee as f64 * asset_index_price) as u64 + fixed_fee;
+        ensure_sdk!(
+            wallet_asset_balance >= amount_sat + fee_sat,
+            PayjoinError::InsufficientFunds
+        );
+
+        let fee = asset_metadata.amount_from_sat(fee_sat);
+        debug!("Estimated payjoin server fee: {fee} ({fee_sat} satoshi units)");
+
+        Ok(fee)
+    }
+
+    async fn build_payjoin_tx(
+        &self,
+        recipient_address: &str,
+        asset_id: &str,
+        amount_sat: u64,
+    ) -> PayjoinResult<(Transaction, u64)> {
+        let fee_asset = AssetId::from_str(asset_id)?;
+        let wallet_utxos = self
+            .onchain_wallet
+            .asset_utxos(&fee_asset)
+            .await?
+            .iter()
+            .map(Utxo::from)
+            .collect::<Vec<_>>();
+        ensure_sdk!(!wallet_utxos.is_empty(), PayjoinError::InsufficientFunds);
+
+        let address = Address::from_str(recipient_address).map_err(|e| {
+            PayjoinError::generic(format!(
+                "Recipient address {recipient_address} is not a valid ElementsAddress: {e:?}"
+            ))
+        })?;
+        let recipients = vec![Recipient {
+            address,
+            asset_id: fee_asset,
+            amount: amount_sat,
+        }];
+
+        let start_request = Request::Start(StartRequest {
+            asset_id: asset_id.to_string(),
+            user_agent: "breezsdk".to_string(),
+            api_key: Some(
+                "97fb6a1dfa37ee6656af92ef79675cc03b8ac4c52e04655f41edbd5af888dcc2".to_string(),
```

Comment:
Do we want to add a config field for that?

Reply:
Done [29b8e3d](https://github.com/breez/breez-sdk-liquid/pull/779/commits/29b8e3d8fb9ffc055d807560ebcb92011bec9546)